includes/Modules/Analytics_4.php

```diff
@@ -387,14 +433,15 @@ protected function get_datapoint_definitions() {
 		}
 
 		if ( Feature_Flags::enabled( 'newsKeyMetrics' ) ) {
-			$datapoints['POST:create-custom-dimension'] = array(
+			$datapoints['POST:create-custom-dimension']         = array(
 				'service'                => 'analyticsdata',
 				'scopes'                 => array( Analytics::EDIT_SCOPE ),
 				'request_scopes_message' => __( 'You’ll need to grant Site Kit permission to create a new Analytics 4 custom dimension on your behalf.', 'google-site-kit' ),
 			);
-			$datapoints['POST:sync-custom-dimensions']  = array(
+			$datapoints['POST:sync-custom-dimensions']          = array(
 				'service' => 'analyticsadmin',
 			);
+			$datapoints['POST:custom-dimension-data-available'] = array( 'service' => '' );
```

Comment:
I was a bit hesitant about specifying this datapoint with an empty service, but we are doing the same for the AdSense `GET:notifications` endpoint so I think it's OK:

https://github.com/google/site-kit-wp/blob/61a101f8cbeb60040d5111009b410065a7be67b9/includes/Modules/AdSense.php#L285

Note that I've not made the endpoint shareable as specced in the IB, this is because it's not possible to resolve the gathering data state on the view-only Dashboard due to needing to retrieving the property from the admin API in order to do so, which can only be done by an authenticated user. This is similar to the GA4 gathering data state which also can't be resolved on the view-only Dashboard for the same reason.

Reply:
I had initially tried it without the `service` key, so just an empty array, but I found this check then failed:

https://github.com/google/site-kit-wp/blob/5aafc62bfa9449790264593dba048da530ee54a7/includes/Core/Modules/Module.php#L292-L294

Having already made a few unplanned changes (I'm thinking of the changes to the report option validation which we've now rolled back) I was hesitant to add another one and update this condition to allow an empty array, so I took the approach taken for the AdSense `GET:notifications` instead.

What do you think, shall we keep it like this or update the condition in `Module.php` to allow an empty array?

Reply:
Yeah, let's mot change that for now. Maybe create a ticket to get back to it later, but that is definitely out of scope for the current task.